astropy/table/tests/test_table.py

```diff
@@ -4,6 +4,7 @@
 import gc
 import sys
 import copy
+import h5py
```

Comment:
I think you'll need to move this test to `io/misc/tests/test_hdf5.py` - this in particular since `h5py` and `pyyaml` are optional dependencies for astropy, i.e., we cannot count on them being present (the tests in `test_hdf5` will show how to account for that - basically you'll need `@pytest.mark.skipif('not HAS_H5PY or not HAS_YAML')`